src/main/java/dal/fatih/todoly/TaskManager.java

```diff
@@ -1,155 +1,262 @@
+        System.out.print("Due Date yyyy-MM-dd (*): ");
+        String lastDateInput = scn.nextLine();
 
+        int counter = 0;
         try {
-            List<Task> foundTask = new ArrayList<>();
-            System.out.println("Last Date");
-            String lastDateInput = scn.nextLine();
-            Date lastDate = dueDateParser.parse(lastDateInput);
-
-            for (Task task : tasks.values()) {
-                if (task.getDate().before(lastDate)) {
-                    foundTask.add(task);
-                }
+            connection = dbConnection.getConnection();
+            Date date=Date.valueOf(lastDateInput);
+            String sql = "select * from TASKS where DUEDATE BETWEEN now() and ?";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setObject(1, date);
+            resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                counter++;
+                String title = resultSet.getString(3);
+                Date duedate = resultSet.getDate(5);
+                System.out.println("Title: " + title);
+                System.out.println("Due Date: " + duedate);
+                System.out.println("-------------------------------------------------");
             }
-            if (foundTask.isEmpty()) {
+            connection.close();
+            preparedStatement.close();
+            resultSet.close();
+            if (counter < 1) {
                 System.out.println("No task found in this date range");
-            } else {
-                System.out.println(foundTask);
             }
-        } catch (Exception e) {
+        } catch (IllegalArgumentException e) {
             System.out.println("Incorrect date format");
+        } catch (SQLException e) {
+            System.out.println("Check database connections, drivers!!!");
         }
     }
 
     private void filterTasksbyNameAndDescription() {
-        List<Task> foundTasks = new ArrayList<>();
-        System.out.println("Word to search");
+        System.out.print("Word to search : ");
         String searchingWord = scn.nextLine();
+        int counter = 0;
 
-        for (Task task : tasks.values()) {
-            if (task.getTitle().toLowerCase().contains(searchingWord.toLowerCase()) || task.getDescription().toLowerCase().contains(searchingWord.toLowerCase())) {
-                foundTasks.add(task);
+        try {
+            connection = dbConnection.getConnection();
+
+            String sql = "select taskId,title,description,dueDate from TASKS where title like ?" + "or description like ?";
+            preparedStatement = connection.prepareStatement(sql);
+            preparedStatement.setObject(1, "%" + searchingWord.toLowerCase() + "%");
+            preparedStatement.setObject(2, "%" + searchingWord.toLowerCase() + "%");
+            resultSet = preparedStatement.executeQuery();
+
+            while (resultSet.next()) {

```

Comment:
Same here. `filterByTitleOrDescription(String keyword)` that would return list of `Task`s